File: Snippets/Reverse/rev.js
//
// Given a string 'str', return its reverse
//

function reverse (str) {
    return (str . split ("") . reverse () . join (""))
}


//
// Given a number 'num', return its reverse
//
function reverse (num) {
    let rev = 0
    while (num > 0) {
        rev *= 10
        rev += num % 10
        num  = Math . floor (num / 10)
    }
    return (rev)
}

File: Snippets/Reverse/rev.go
//
// Given a string 'str', return its reverse
//
func reverse (str string) string {
    runes := [] rune (str)
    for i, j := 0, len (runes) - 1; i < j; i, j = i + 1, j - 1 {
        runes [i], runes [j] = runes [j], runes [i]
    }
    return (string (runes))
}

//
// Given a number 'num', return its reverse
//
func reverse (num int) int {
    rev := 0
    for num > 0 {
        rev *= 10
        rev += num % 10
        num /= 10
    }
    return rev
}
